addons/binding/org.openhab.binding.unifi/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="unifi"
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+		xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="controller">
+
+		<label>UniFi Controller</label>
+		<description>A UniFi controller.</description>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>Hostname</label>
+				<description>Hostname of IP address of the UniFi Controller</description>
+				<default>unifi</default>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="integer" max="65535" min="1" required="true">
+				<label>Port</label>
+				<description>Port of the UniFi Controller</description>
+				<default>8443</default>
+			</parameter>
+			<parameter name="username" type="text" required="true">
+				<label>Username</label>
+				<description>The username to access the UniFi Controller.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>The password to access the UniFi Controller.</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="refresh" type="integer" required="false">
+				<label>Refresh Interval</label>
+				<description>The refresh interval in seconds to poll the UniFi controller</description>
+				<default>10</default>
+			</parameter>
+			<parameter name="considerHome" type="integer" required="false">
```

Comment:
> I'm not sure whether this should be part of the binding or is a generic concept, @kaikreuzer wdyt?

@kaikreuzer, see my question above 🙂

Reply:
While I do believe this is probably better suited as a generic concept in the OH framework, I feel this parameter needs to stay until it can be refactored to a higher level. The way the UniFi controller caches the connected clients, we really need a way to force the binding to mark a client as offline even though they still appear in the UniFi controller. Without it, we get long delays in marking a client offline, which is undesirable and leads users to think the binding is not working properly.

I will happily refactor this if the OH core ever provides a generic "considerHome" parameter.

Reply:
I believe I had something else in my mind, this timeout parameter does not seem to be a problem, sorry